packages/@react-spectrum/menu/docs/MenuTrigger.mdx

```diff
@@ -0,0 +1,439 @@
+
+The example below uses `onOpenChange` to update a separate span element with the current open state of the Menu.
+
+```tsx example
+function Example() {
+  let [state, setState] = React.useState(false);
+
+  return (
+    <div>
+      <MenuTrigger onOpenChange={(isOpen) => setState(isOpen)}>
+        <ActionButton>
+            Menu Button
+        </ActionButton>
+        <Menu items={[{name: 'Aardvark'}, {name: 'Kangaroo'}, {name: 'Snake'}]} itemKey="name">
+          {item => <Item>{item.name}</Item>}
+        </Menu>
+      </MenuTrigger>
+      <span style={{'margin-left': '8px'}}>Current open state: {state.toString()}</span>
+    </div>
+  );
+}
+```
+
+Menu accepts an `onAction` handler which is triggered whenever a Menu item is selected.
+
+The example below uses `onAction` to tell the user which Menu item was selected.
+
+```tsx example
+function Example() {
+  let onAction = (value) => alert('Menu item selected: ' + value);
+
+  return (
+    <div>
+      <MenuTrigger>
+        <ActionButton>
+            Menu Button
+        </ActionButton>
+        <Menu items={[{name: 'Aardvark'}, {name: 'Kangaroo'}, {name: 'Snake'}]} itemKey="name" onAction={onAction}>
+          {item => <Item>{item.name}</Item>}
+        </Menu>
+      </MenuTrigger>
+    </div>
+  );
+}
+```
+
+## Props
+
+### MenuTrigger
+
+<PropTable component={docs.exports.MenuTrigger} links={docs.links} />
+
+### Menu
+
+<PropTable component={docs.exports.Menu} links={docs.links} />
+
+## Visual Options
+
+### Align and Direction
+
+[View guidelines](https://spectrum.adobe.com/page/popovers/#Placement)
+
+```tsx example
+<MenuTrigger align="start" direction="bottom" shouldFlip={false}>
+  <ActionButton>
+      align=start direction=bottom
+  </ActionButton>
+  <Menu items={[{name: 'Aardvark'}, {name: 'Kangaroo'}, {name: 'Snake'}]} itemKey="name">
+    {item => <Item>{item.name}</Item>}
+  </Menu>
+</MenuTrigger>
+<MenuTrigger align="start" direction="top" shouldFlip={false}>
+  <ActionButton>
+      align=start direction=top
+  </ActionButton>
+  <Menu items={[{name: 'Aardvark'}, {name: 'Kangaroo'}, {name: 'Snake'}]} itemKey="name">
+    {item => <Item>{item.name}</Item>}
+  </Menu>
+</MenuTrigger>
+<MenuTrigger align="end" direction="bottom" shouldFlip={false}>
+  <ActionButton>
+      align=end direction=bottom
+  </ActionButton>
+  <Menu items={[{name: 'Aardvark'}, {name: 'Kangaroo'}, {name: 'Snake'}]} itemKey="name">
+    {item => <Item>{item.name}</Item>}
+  </Menu>
+</MenuTrigger>
+<MenuTrigger align="end" direction="top" shouldFlip={false}>
+  <ActionButton>
+      align=end direction=top
+  </ActionButton>
+  <Menu items={[{name: 'Aardvark'}, {name: 'Kangaroo'}, {name: 'Snake'}]} itemKey="name">
+    {item => <Item>{item.name}</Item>}
+  </Menu>
+</MenuTrigger>
+```
+
+### Autofocus
+
+Applying `autoFocus` to the Menu of the MenuTrigger sets focus within the Menu upon opening.
+
+```tsx example
+<MenuTrigger>
```

Comment:
this is again very hard to read, maybe we should split all these up into separate examples with titles?